source/common/router/router.cc

```diff
@@ -965,15 +966,21 @@ void Filter::onUpstreamAbort(Http::Code code, StreamInfo::ResponseFlag response_
 
     callbacks_->streamInfo().setResponseFlag(response_flags);
 
-    callbacks_->sendLocalReply(
-        code, body,
-        [dropped, this](Http::ResponseHeaderMap& headers) {
-          if (dropped && !config_.suppress_envoy_headers_) {
-            headers.setReferenceEnvoyOverloaded(Http::Headers::get().EnvoyOverloadedValues.True);
-          }
-          modify_headers_(headers);
-        },
-        absl::nullopt, details);
+    try {
+      callbacks_->sendLocalReply(
```

Comment:
Do we need this for other sendLocalReply functions?  Should it be in the HCM function instead, or have you checked to make sure the others are all driven by downstream data?

Reply:
I think you have a really good point here.  I did a search for other local reply cases that are outside the call stack of decodeHeaders and found more that I was hoping for.

Reply:
Yeah I would definitely put this inside the HCM for now. Per our video chat about removing exceptions from this code, I think this is a case in which we would swallow the error and reset/disconnect downstream on the next dispatch. cc @asraa @yanavlasov

Reply:
Changed to do the try/catch in sendLocalReply and reset the stream from there.

mattklein123: is there prior art for unrolling the stack and resetting on next dispatch?  I have worries about further work being done on the connection after the rejected sendLocalReply.

I do have concerns about reset in arbitrary deep stacks also, perhaps flood protection should be rethought slightly: instead of doing it on encode, can we do it during decode?  Decode is always done in a relatively shallow call stack that is protected by an existing try/catch block.

Reply:
Yeah this makes me cringe (not your fault). If we go with this we should definitely make sure it's clear this is going to be removed once we do the exception removal from the codecs.

> mattklein123: is there prior art for unrolling the stack and resetting on next dispatch? I have worries about further work being done on the connection after the rejected sendLocalReply.

I don't think we do this today, but we do track whether we are currently dispatching in the H/2 codec via `dispatching_`. So it would not be too difficult to extend this to keep track of state that we want to do on the next dispatch.

> I do have concerns about reset in arbitrary deep stacks also, perhaps flood protection should be rethought slightly: instead of doing it on encode, can we do it during decode? Decode is always done in a relatively shallow call stack that is protected by an existing try/catch block.

Yes exactly this is what I had in mind also. Either we always do it in the context of decode and never encode, or maybe we do it in encode, but never block anything and only mark a connection as flooded to be terminated ASAP on the next decode. Maybe the two of these are the same thing, but this is what I had in mind in general.

Reply:
Doing the close in the deep stack also makes me cringe.

I'm going to try to move flood protection to decode.  If it works out, I'll send it out as a separate PR that would replace this one.